src/main/java/uk/nhs/digital/nhsconnect/nhais/model/edifact/Section.java

```diff
@@ -0,0 +1,55 @@
+package uk.nhs.digital.nhsconnect.nhais.model.edifact;
+
+import lombok.Getter;
+import uk.nhs.digital.nhsconnect.nhais.model.edifact.message.MissingSegmentException;
+import uk.nhs.digital.nhsconnect.nhais.model.edifact.message.ToEdifactParsingException;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.function.Supplier;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+public abstract class Section {
+    @Getter
+    private final List<String> edifactSegments;
+
+    public Section(List<String> edifactSegments) {
+        this.edifactSegments = edifactSegments;
+    }
+
+    protected List<String> extractSegments(String key) {
+        return edifactSegments.stream()
+            .map(String::strip)
+            .filter(segment -> segment.startsWith(key))
+            .collect(Collectors.toList());
+    }
+
+    protected Optional<String> extractOptionalSegment(String key) {
```

Comment:
Will this need to be Optional<List<String>> for repeated optional segments? Or just List<String> and empty if none?

Reply:
I see this is the optional version of "extractSegment" 👍